tests/tools/FFI/bluechi-tester

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/python
+
+import argparse
+import sys
+
+from dasbus.typing import get_variant, Tuple
+from dasbus.connection import AddressedMessageBus, InterfaceProxy, ObjectProxy
+from dasbus.error import DBusError
+
+from gi.repository import GLib
+from typing import Any
+
+OBJECT_PATH_BLUECHI = "/org/eclipse/bluechi/internal"
+OBJECT_PATH_AGENT = OBJECT_PATH_BLUECHI + "/agent"
+
+INTERFACE_BLUECHI = "org.eclipse.bluechi"
+INTERFACE_AGENT = INTERFACE_BLUECHI + ".internal.Agent"
+
+SUPPORTED_SIGNALS = ["JobDone"]  # format: ["SignalName1, SignalName2, etc"]
+
+
+class BlueChiTester(object):
+
+    def __init__(self, node_name: str) -> None:
+        self.node_name: str = node_name
+        self.peer_bus: AddressedMessageBus = None
+        self.peer_bus_proxy: (InterfaceProxy | ObjectProxy | Any) = None
+
+    def run(self, timeout=5):
+        loop = GLib.MainLoop()
```

Comment:
I noticed that without the loop we cannot wait the signals. In fact, without the loop the program end before the signals answer back, example, from 10 we see only 5 answering and program ends.

Reply:
You are right - removing the loop results in not all 10 signals reaching the controller, that is interesting. 
However, I noticed adding a `time.sleep(1)` also seems to ensure that all signals are sent. Looking at the [GLib timeout_add_seconds documentation](https://docs.gtk.org/glib/func.timeout_add_seconds.html), the callback `_kill_loop` is executed repeatedly after a specific interval - and since we kill the loop on the first time, its basically just a sleep. 
I know this is nitpicky, but could we replace the loop code with a simple `time.sleep()`?

Reply:
I tried that on the weekend and doesn't help, imagine if we scale to 100000 calls, we will need at least 0.8 sleep * 100000 to make sure it will return back. Doesn't seem correct.

Reply:
You meant:
```
diff --git a/tests/tools/FFI/bluechi-tester b/tests/tools/FFI/bluechi-tester
index 3b45cc6..e1b9964 100755
--- a/tests/tools/FFI/bluechi-tester
+++ b/tests/tools/FFI/bluechi-tester
@@ -31,12 +31,12 @@ class BlueChiTester(object):
             signal_name: str,
             num_signals: int):

-        loop = GLib.MainLoop()
+        #loop = GLib.MainLoop()

-        def _kill_loop():
-            loop.quit()
-            print("\nDone!")
-            sys.exit(0)
+        #def _kill_loop():
+        #    loop.quit()
+        #    print("\nDone!")
+        #    sys.exit(0)

         parameters = get_variant(
                 Tuple[int, str],
@@ -59,8 +59,8 @@ class BlueChiTester(object):
             print(f"  * Sending {signal_name} {number}...")

         # Schedule a callback to exit the EventLoop after the timeout
-        GLib.timeout_add_seconds(timeout, _kill_loop)
-        loop.run()
+        #GLib.timeout_add_seconds(timeout, _kill_loop)
+       # loop.run()
```

Reply:
The `time.sleep(1)` at the end is missing. Added a suggestion (just for demo purposes): https://github.com/containers/bluechi/pull/614/files#r1369122580